html/app.js

```diff
@@ -61,16 +80,10 @@ async function force_permission_prompt() {
 }
 
 async function wait_for_mic_permissions() {
-  try {
```

Comment:
Are you asserting that `navigator.permissions.query` can't fail here, or are you letting it throw, intending to catch it somewhere else? (It can definitely fail. I think in Firefox it may fail every time? https://bugzilla.mozilla.org/show_bug.cgi?id=1449783)

Reply:
I guess in particular, in Firefox it will throw even though nothing is wrong, and the only thing you can do is continue to the permission prompt.